### DISCUSSION TITLE
Call for alpha testers and Discord server announcement

### DISCUSSION THREAD

**Discussion by dinki** [dinki/View-Assist#10] [Announcements]
I am once again humbled by the interest in the View Assist project.  I have had a bunch of things holding back time spent on VA with end of year school activities and major projects at work.  Those are clearing and I should be able to spend more time with the VA project.

Some may have noticed that I have been populating the wiki with Youtube videos in support of the written documentation.  The channel has gained about 75 subscribers without much more than a handful of how tos on getting things prepped for Stream ASsist use.  Wow!   As such, I am getting very close to having everything in place for the release of the View Assist portion of the system.  

I am hesitant to release View Assist as it stands now as it is somewhat advanced in terms of the complexity of set up.  By that I mean there's a bit of YAML needed to add to the configuration.yaml and some helpers that need to be created per device.  The rest is setting up scripts and automations.  I recently discovered pyscript which might help simplify these things.  This is next step work for me to ensure that this is the right way to go.

I am looking for folks who are willing to do some alpha testing.  I can provide instructions on how to set things up as they are now and to gauge your feedback on just how difficult it is.  I'd really like to make the barrier to using VA as low as possible to try and offer it to as many as possible.  I think that by getting a small group of testers it should be easier for me to help and possibly you guys can help each other too.  I don't think it will be difficult for most of you.

I've set up a Discord server for those that are interested and I will tag some of you that have already been active in the discussion area.  It is definitely not limited to just those tagged and I encourage anyone interested in joining.  I am US based so will be most available during waking hours but will follow-up on any missed comments/questions as I can.

Check out the server here https://discord.gg/FTzF4cKb

Thanks again for the continued interest and help!

@mngarchow @kilthor @pgale @mllockwood @mrtango @danstennett @ekashuba  @bakanode @flight-lab

**Comment by pgale** [comment]
For the thinksmart view, is this just on A11? I seem to remember there's a reason for this - probably audio related.

**Reply by dinki** [reply]
For now, yes.  I am thinking of getting a third unit to test other ROMs but I can certainly say that I could not get audio working using the A8 ROM.  I really just don't have enough time to test things at the moment.   The A11 ROM definitely has issues but it is mostly stable so sticking with it for now.

**Reply by pgale** [reply]
Ok - I have four units running 8.1 in production around the house and the fifth unit is my test device. I just couldn't get it to boot on A11 no matter what I tried. I do like the idea of VA though so I will have a think about swapping a unit over and trying A11 again. If I do, I'm happy to test. I'll have a think over the next few days. There's a fair bit of work to swap one out though.

**Reply by dinki** [reply]
I'll admit that I know a lot more now about the TS than I did when I tested A8 .  I'd be interested if you could try to get browermod to work with A8.  Particularly audio.  I could not get it to work at all.

**Reply by pgale** [reply]
I did get browsermod working and tried the available webviews but none enabled audio through the browser unfortunately. Is that what you were looking for?

**Reply by dinki** [reply]
Yes. Thanks for confirming. 

Do you know how many different ROMs exist for the TS?

**Reply by pgale** [reply]
Well there's A8.1 A11 and PMOS (Linux) and of course all of the original versions with Teams or Zoom. Someone did mention A10 that had been discussed on Discord but I never tracked that down.

**Comment by BakaNode** [comment]
I used to run HA but dropped it for HomeBridge due to HA constantly killing itself about every 3-4 months like clockwork.

I found this project while researching how to update the Android version for the View and would love to help with testing newer or more complete versions of Android. I have several apps that don’t work with the current A11 rom.

**Reply by dinki** [reply]
Wow.  That is super odd.  I've run HA for about seven years now on a few different machines and never  had issues like this.  Would certainly like to help troubleshoot if you want to give it another go.  What device were you running HA on?

**Reply by BakaNode** [reply]
My main use case for this device seems to all require a newer version of Android or at least a more complete version.

What makes matters worse is the core function of Teams doesn’t even work anymore.

Is there any way to flash it back to stock or can you point me in the right direction for a newer rom?

**Comment by pgale** [comment]
Now I have A11 running again - I'm in for the testing. Let me know what you want me to do...

**Comment by ADHDSquir** [comment]
@dinki, heads up, I saw your request over on the StreamAssist repo to add area awareness to assist pipelines started from StreamAssist and I submitted a pull request that adds that functionality earlier this morning. I don't use custom sentences so I couldn't test that exact use case but I'm fairly certain it will apply there as well.

I'm using 8.1 without many issues. What part of the audio was giving you trouble? I'm debugging the case where some TTS responses play at max volume, but that's the only issue I've had.

This is your project, for which we are all grateful, so roll things out however you wish. But I think the beauty of GitHub is you can publish what you have in a dev branch and those of us who are along for the ride can not just report issues for you to fix but also help implement fixes via pull requests.

**Reply by dinki** [reply]
I cannot tell you how excited I am to hear this and I will be trying this out tomorrow!  This will simplify things GREATLY and will allow for me to easily integrate the SA satellites with the ESP32 ones.  So awesome!

RE: 8.1 audio, I was not able to get audio working using Browsermod which is a critical component for VA.  It is used to expose the media player device as well the View switching.

It's a great idea to set up dev branch.   I will work towards this for sure.

 I've used GH for years but never in this manner so advice is certainly appreciated.  The VA system is really a combination of a whole bunch of integrations, custom configuration, scripts, automations, views, etc so figuring out a logical way to present this stuff is challenging for me.  I looked at creating an integration but that appears to be above my current skill level.

**Reply by dinki** [reply]
@jshtz4 , I thought I could just pull the changed file (sensor.py) that you created and drop it in as a replacement for my version in custom_components/streamassist directory and restart to apply.  SA is showing unavailable sensors when I tried doing that.   

Can you tell me the best way to try your fix without having to wait for it to get rolled into a release?  Thanks again!

**Reply by ADHDSquir** [reply]
Sensor.py and __init__.py in PR #26 correct warnings from Home Assistant that appeared with 2024.5. PR #27 adds the device_id for area awareness and changes switch.py. You should be able to just do a drop-in replacement in your custom_components directory. Maybe you have an older version of the codebase? Also, you have to restart home assistant, not just the integration. I wasn't able to find a way to get changes to take effect without a complete reboot.

Did you use the GUI for all your stream_assist configs or did you do some via a yaml service?

**Reply by dinki** [reply]
I'm running v2.0.0 installed via HACS.  Seems current.  All config done through GUI.  For View Assist I am thinking of 'simplifying' the custom sensor config by creating an automation that calls a service to create these but that's still in my head.

I missed that first PR you mentioned.  I'll try to grab that extra file (init.py) and see if that does the trick.  

*EDIT*  Confirmed working!  You are a superstar!  This will definitely simplify things.  Awesome!

**Reply by ADHDSquir** [reply]
The pull requests shows the commits and exactly what was changed. If you want to just grab the modified version of a file that's part of a pull request or commit, at the top of the screen you'll see a line like "jshtz4 wants to merge 1 commit into AlexxIT:main from jshtz4:device_id-to-assist" where the names of the two users and branches are links to the repositories. So you can click on jshtz4:device_id-to-assist and you'll be taken to my copy of the repo where everything is identical to AlexxIT's repo except for the file I changed.

Glad it's working for you!
